### PR TITLE
refactor: use theme-ui instead of styled-components & friends

### PR DESCRIPTION
**Pull request by luniki**
Currently we are using a mix of styled-components with styled-system. As esp. the latter is quite opinonated and theming was harder than it should be, I refactored the code base to use theme-ui (see: https://theme-ui.com/ )

At first I tried to convert the text heavy pages like about and impressum into MDX but this did not work out that easily. Therefore I am delaying that for further investigation.

The rest of the refactorings is straight forward and more or less a transform of what we had before.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -31,7 +34,9 @@ function App(props) {
   const { t, i18n } = useTranslation();
 
   const [chosenLanguage, setChosenLanguage] = useState('en');
-  const [isModalOpen, setModalOpen] = useState(!window.localStorage.getItem(UPLOAD_SETTINGS_KEY));
+  const [isModalOpen, setModalOpen] = useState(
+    false && !window.localStorage.getItem(UPLOAD_SETTINGS_KEY)
```
Comment (lkiesow) [inline src/App.js]: Doesn't this just evaluate to always `false`?

File: src/theme.js [hunk 2]
```diff
@@ -1,4 +1,62 @@
-export default {
+// export default {
```
Comment (lkiesow) [inline src/theme.js]: If it's all commented out, it can go, can't it?

**Comment by lkiesow** [comment]
The new UI seems to be tested in Firefox. This is how it looks in Chrome:

Screenshot from 2019-11-06 11-15-42

**Comment by lkiesow** [comment]
This is how the new UI looks on a mobile Chrome. It does not seem to detect the camera any longer:

__0b622bef-0522-48e2-859a-525ed17ca718

…and for this message, some padding would be nice.

---

On the same system, in Firefox, you get the option for a camera but it doesn't work, plus, the camera icon seems to be unintentionally moved slightly off the center.

__f257f9f7-0eb1-4c7b-b002-eb39ed4a35f7

**Comment by lkiesow** [comment]
These short-lived notification pop-ups are an accessibility nightmare:

Screenshot from 2019-11-06 11-36-52

If you need visual aid, you are likely to miss them.

**Comment by lkiesow** [comment]
Last (for this round of testing) and certainly only a minor thing, the new bright blue buttons look odd on the otherwise more monochrome application. I suggest staying with gray-tones or if necessary, at least use less agressive colors:

Screenshot from 2019-11-06 11-36-30

File: src/App.js (hunk 1 above)
Comment (luniki) [reply src/App.js]: Argh, that is a leftover after a broken develop.opencast.org. I'll remove this.

File: src/theme.js (hunk 2 above)
Comment (luniki) [reply src/theme.js]: True. Originally I had another theme but I guess they are now nicely coalesing. I will remove the comment.

**Comment by luniki** [comment]
* The Chrome problem will be resolved. 
* I have no idea why the webcam does not show up in Chrome but will investigate. Meanwhile I restyled the error text.
* The icon is centered but kind of an optical illusion.
* I will change the toast for missing input fields. 
* The buttons get another color.

**Comment by luniki** [comment]
* Chrome < v72 does not conform to the spec and supports only navigator.getDisplayMedia instead of navigator.mediaDevices.getDisplayMedia

**Comment by luniki** [comment]
This should take care of most of them. There is still no solution for detecting support of getDisplayMedia/getUserMedia in certain (mobile) browsers. We should talk about strategies of non-disruptive detecting support.

**Comment by lkiesow** [comment]
> * Chrome < v72 does not conform to the spec and supports only navigator.getDisplayMedia instead of navigator.mediaDevices.getDisplayMedia

I've tested on way newer versions:

- Google Chrome
   - 78.0.3904.87 (Official Build) (64-bit)
- Revision
   - 20c21f4010010f32462ea8e1d6af30cef66d48c8-refs/branch-heads/3904@{#840}
- OS
   - Linux

and 

- Google Chrome
   - 78.0.3904.62 (Official Build) (32-bit)
- Revision
   - 966ec96e814edd1cf831a1d48b66b6d843ed4334-refs/branch-heads/3904@{#781}
- OS
  - Android 9; ONEPLUS A5000 Build/PKQ1.180716.001
- Google Play services
   - SDK=16798000; Installed=19530037; Access=1pJavaScriptV8 7.8.279.17

**Comment by lkiesow** [comment]
Still, no luck testing this on mobile devices but I'm wondering if that is caused by not deploying via HTTPS. Did you manage to run this on a mobile device without?

A visual issue is still present when displaying portrait-mode recordings. Though that's not new I'm not sure if this wasn't meant to address this issue as well:

Screenshot from 2019-11-12 00-35-21

And finally, there are some merge conflicts.

**Comment by luniki** [comment]
So are there still issues with the refactoring? Or are these issues that existed before?

**Pull request merged**